### PR TITLE
Pro 3109 heroku tut

### PR DESCRIPTION
**Pull request by BoDonkey**
## Summary

*Summarize the changes briefly, including which issue/ticket this resolves. If it closes an existing Github issue, include "Closes #[issue number]"*

This PR adds a new recipe for hosting an apostrophe site on Heroku. It is a port of the a2 recipe with updates. Closes PRO-3109.

## What are the specific steps to test this change?

*For example:*
> 1. Run the website and log in as an admin
> 2. Open a piece manager modal and select several pieces
> 3. Click the "Archive" button on the top left of the manager and confirm that it should proceed
> 4. Check that all pieces have been archived properly

1. Build the documentation
2. Navigate to `cookbook/deploying-to-heroku.html`
3. 
## What kind of change does this PR introduce?
*(Check at least one)*

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [X] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**
I'm not sure how much of the section after "Victory!" is needed and/or accurate. I'm not very sure about CORS

### CODE REVIEW

File: docs/cookbook/deploying-to-heroku.md [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+
+## Deploying Apostrophe to Heroku
+
+[Heroku](http://heroku.com) is a great starting point for cloud hosting because it is simple to set up, but all of the cloud's challenges come into play. What we learn by deploying to Heroku can be applied equally to Amazon EC2, Microsoft Azure, and other cloud hosting services.
+
+To reduce costs for this how-to, we'll be using free services from Amazon Web Services and MongoDB Atlas, a MongoDB cloud hosting service from the creators of MongoDB. But remember that you can choose paid plans with much higher capacity and performance. Everything in this tutorial is designed to scale smoothly to those paid offerings. Heroku's offerings are in flux. They are eliminating their free tier, but they are offering low-price 'Eco' and 'Basic' plans that have the same provisioning as their free tier that was used to write this tutorial.
```
Comment (boutell) [inline docs/cookbook/deploying-to-heroku.md:10]: Interesting, didn't know that.

as their free => as the free (better grammar? I think?)

File: docs/cookbook/deploying-to-heroku.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+MongoDB Atlas requires us to allowlist the IP addresses that should be allowed to talk to the database. **Yes, it is secured by an encrypted password,** but this still helps to reduce potential DOS attacks.
+
+This is a problem with Heroku because it may connect from many IP addresses.
+
+If you are buying a larger Atlas plan you may be able to use the "VPC Peering" option, the details of which are beyond the scope of this recipe. Otherwise, use this IP address range:
+
+`0.0.0.0/0`
+
+This will allow you to connect for the purposes of this tutorial, but should be modified for a production site. One solution is to use the [fixie-socks](https://elements.heroku.com/addons/fixie-socks) add-on from Heroku.
```
Comment (boutell) [inline docs/cookbook/deploying-to-heroku.md:87]: Honestly this is what we do for production too. I think you don't have to word it quite as strongly as "should be modified." Using fixie-socks could impact performance... there are pros and cons.

File: docs/cookbook/deploying-to-heroku.md [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+### Telling Heroku about your database
+
+You will need to set an environment variable in Heroku so that your dynos can communicate with the database.Within your new database deployment, click on the 'connect' button next to the name. Next, click on 'Connect your application'. By default, the driver drop-down menu should be set to 'Node.js' and version '4.1 or later'. If they aren't, change thhem and then copy the connection string. Paste the string into a text editor and add your password. Again, make sure to URL encode any special special characters. Now use this string to set set a environment variable in heroku. It should look something like this:
```
Comment (boutell) [inline docs/cookbook/deploying-to-heroku.md:91]: database.Within -> database. Within

Comment (boutell) [reply docs/cookbook/deploying-to-heroku.md:91]: tthem -> them

Comment (boutell) [reply docs/cookbook/deploying-to-heroku.md:91]: "special special" is very special 😄

File: docs/cookbook/deploying-to-heroku.md [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+```
+$ heroku config:set 'APOS_MONGODB_URI=mongodb+srv://<yourUserName>:<password>@heroku-tut.lrzxt0l.mongodb.net/?retryWrites=true&w=majority)'
```
Comment (boutell) [inline docs/cookbook/deploying-to-heroku.md:94]: No `)`

**Review by boutell — CHANGES_REQUESTED** [review]
I don't see the asset build task being run anywhere in this tutorial. We do need to make sure Heroku does that so the site isn't unreachable while it builds assets *after* startup.

We should also set `NODE_ENV` to `production` on Heroku (after which you'll find relying on the first startup to create the assets for us doesn't cut it).

**Comment by boutell** [comment]
* Old asset builds will not be automatically cleaned up with A3, but it also doesn't really matter for the most part, so you can drop that mention for now.
* Without `APOS_UPLOADFS_ASSETS=1` being set in Heroku it is not true that build artifacts will get copied to S3 for efficient delivery.
* Please investigate whether and how Heroku still supports "build steps" to do things like the asset build for you so you can just tweak release-id and git push.

File: docs/cookbook/deploying-to-heroku.md (hunk 1 above)
Comment (BoDonkey) [reply docs/cookbook/deploying-to-heroku.md:10]: Changed.

File: docs/cookbook/deploying-to-heroku.md (hunk 2 above)
Comment (BoDonkey) [reply docs/cookbook/deploying-to-heroku.md:87]: Changed, but I could be persuaded to eliminate it.

File: docs/cookbook/deploying-to-heroku.md (hunk 3 above)
Comment (BoDonkey) [reply docs/cookbook/deploying-to-heroku.md:91]: Changed along with 'set set' the set of all sets!

File: docs/cookbook/deploying-to-heroku.md (hunk 4 above)
Comment (BoDonkey) [reply docs/cookbook/deploying-to-heroku.md:94]: Changed.

File: docs/cookbook/deploying-to-heroku.md [hunk 5]
```diff
@@ -110,7 +115,7 @@ Press Control-C after you successfully test the site. Startup may take an extra
 
 > If you do not run `npm start` with the environment variable set correctly, it'll seem to work because it will connect to your own mongodb. You can shut down your local mongodb server temporarily if you want to be really, really sure.
 
-Now Your database exists in the cloud, but it doesn't contain any users, so you won't be able to log in. Let's use the command line to connect again to fix that:
+Now Your database exists in the cloud, but it doesn't contain any users, so you won't be able to log in. So, let's use the command line to connect again to fix that:
```
Comment (boutell) [inline docs/cookbook/deploying-to-heroku.md:118]: Now Your -> Now your

**Review by boutell — APPROVED** [review]

**Pull request merged**